File: src/main/java/com/back/mozu/domain/setting/dto/HolidayDto.java
package com.back.mozu.domain.setting.dto;

import com.back.mozu.domain.setting.entity.Holiday;
import jakarta.validation.constraints.NotNull;
import java.time.LocalDate;
import java.util.List;

public class HolidayDto {

    public record HolidayItem(
            LocalDate date,
            String reason
    ) {
        public static HolidayItem from(Holiday holiday) {
            return new HolidayItem(
                    holiday.getDate(),
                    holiday.getReason()
            );
        }
    }

    public record GetHolidaysResponse(
            int totalCount,
            List<HolidayItem> holidays
    ) {
    }

    public record CreateHolidayRequest(
            @NotNull
            LocalDate date,
            String reason
    ) {
    }

    public record CreateHolidayResponse(
            LocalDate date,
            String reason,
            int conflictingReservationCount
    ) {
    }
}

File: src/main/java/com/back/mozu/domain/setting/dto/HolidayDto.kt
package com.back.mozu.domain.setting.dto

import com.back.mozu.domain.setting.entity.Holiday
import jakarta.validation.constraints.NotNull
import java.time.LocalDate

class HolidayDto {
    data class HolidayItem(
        val date: LocalDate,
        val reason: String?
    ) {
        companion object {
            fun from(holiday: Holiday): HolidayItem {
                return HolidayItem(
                    holiday.date,
                    holiday.reason
                )
            }
        }
    }

    data class GetHolidaysResponse(
        val totalCount: Int,
        val holidays: List<HolidayItem> // MutableList -> List, non-null
    )

    data class CreateHolidayRequest(
        @field:NotNull val date: LocalDate,
        val reason: String?
    )

    data class CreateHolidayResponse(
        val date: LocalDate,
        val reason: String?,
        val conflictingReservationCount: Int
    )
}